SkyDrive/SkyDrive.Tests/UnitTests/InstructorServiceTests.cs

```diff
@@ -0,0 +1,277 @@
+            var service = new InstructorService(_mockRepository.Object);
+
+            //Act
+            var result = await service.GetInstructorById(2, CancellationToken.None);
+
+            //Assert
+            _mockRepository.Verify(x => x.GetById(2, CancellationToken.None), Times.Once);
+            Assert.NotNull(result);
+            Assert.IsType<InstructorModel>(result);
+        }
+
+        [Fact]
+        public async Task GetInstructorById_IncorrectId_ReturnNull()
+        {
+            //Arrange
+            _mockRepository.Setup(x => x.GetById(2, CancellationToken.None)).ReturnsAsync(value: null);
+            var service = new InstructorService(_mockRepository.Object);
+
+            //Act
+            var result = await Assert.ThrowsAsync<EntityNotFoundException>(() => service.GetInstructorById(2, CancellationToken.None));
+
+            //Assert
+            _mockRepository.Verify(x => x.GetById(2, CancellationToken.None), Times.Once);
+            Assert.NotNull(result);
+            Assert.Equal("Instructor with id: 2 not found", result.Message);
+        }
+
+        [Fact]
+        public async Task CreateInstructor_CorrectModel_ReturnEventModel()
+        {
+            //Arrange
+            var instructorModel = new InstructorModel
+            {
+                Id = 0,
+                FirstName = "FirstName 2",
+                LastName = "LastName 2",
+                MiddleName = "MiddleName 2",
+                Experience = 7,
+                Events = null
+            };
+            var instructorEntity = new InstructorEntity
+            {
+                Id = 0,
+                FirstName = "FirstName 2",
+                LastName = "LastName 2",
+                MiddleName = "MiddleName 2",
+                Experience = 7,
+                Events = null
+            };
+
+            _mockRepository.Setup(x => x.Create(It.IsAny<InstructorEntity>(), CancellationToken.None)).ReturnsAsync(value: instructorEntity);
+            var service = new InstructorService(_mockRepository.Object);
+
+            //Act
+            var result = await service.CreateInstructor(instructorModel, CancellationToken.None);
+
+            //Assert
+            _mockRepository.Verify(x => x.Create(It.IsAny<InstructorEntity>(), CancellationToken.None), Times.Once);
+            Assert.NotNull(result);
+            Assert.IsType<InstructorModel>(result);
+        }
+
+        [Fact]
+        public async Task UpdateInstructor_CorrectModel_ReturnInstructorModel()
+        {
+            //Arrange
+            var instructorModel = new InstructorModel
+            {
+                Id = 3,
+                FirstName = "FirstName 2",
+                LastName = "LastName 2",
+                MiddleName = "MiddleName 2",
+                Experience = 7,
+                Events = null
+            };
+            var instructorEntity = new InstructorEntity
```

Comment:
Now stay it like that. But in future it would be better to use AutoData and your code will become much shorter